service/controller/v11/resource/pod/delete.go

```diff
@@ -38,80 +41,201 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		}
 	}
 
-	r.logger.LogCtx(ctx, "debug", "found the current version of the reconciled pod in the Kubernetes API")
+	{
+		a := currentPod.GetAnnotations()
+		if a == nil {
+			return microerror.Mask(missingAnnotationError)
+		}
+		v, ok := a[key.AnnotationPodDrained]
+		if !ok {
+			return microerror.Mask(missingAnnotationError)
+		}
+		if v == "True" {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "pod is already drained")
+			resourcecanceledcontext.SetCanceled(ctx)
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+			return nil
+		}
+	}
```

Comment:
The diff is not useful. The implementation is quite straight forward now. Maybe just check the file.